internal/gitinterface/tree.go

```diff
@@ -74,9 +88,53 @@ func (r *Repository) GetMergeTree(commitAID, commitBID Hash) (Hash, error) {
 		return r.GetCommitTreeID(commitBID)
 	}
 
-	stdOut, err := r.executor("merge-tree", commitAID.String(), commitBID.String()).executeString()
+	niceGit, err := IsNiceGitVersion()
 	if err != nil {
-		return ZeroHash, fmt.Errorf("unable to compute merge tree: %w", err)
+		return ZeroHash, err
+	}
+
+	var stdOut string
+
+	if !niceGit {
+		// Older Git versions do not support merge-tree, and, as such, require
+		// quite a long workaround to find what the merge tree is. This
+		// workaround boils down to:
+		// Create new branch > Merge into said branch > Extract tree hash
+		currentBranch, err := r.executor("branch", "--show-current").executeString()
+		if err != nil {
+			return ZeroHash, fmt.Errorf("unable to determine current branch: %w", err)
+		}
+
+		err = r.SetReference(TempMergeComputationRef, commitAID)
+		if err != nil {
+			return ZeroHash, fmt.Errorf("unable to set the merge computation ref: %w", err)
+		}
+
+		_, err = r.executor("checkout", TempMergeComputationRef).executeString()
+		if err != nil {
+			return ZeroHash, fmt.Errorf("unable to create merge computation branch: %w", err)
+		}
+
+		_, err = r.executor("merge", commitBID.String()).executeString()
+		if err != nil {
+			return ZeroHash, fmt.Errorf("unable to perform merge: %w", err)
+		}
+
+		stdOut, err = r.executor("show", "-s", "--format=%P", TempMergeComputationRef).executeString()
```

Comment:
Isn't this listing the parents of the merge commit? Do we have a gap in our tests?